framework/auth/core.py

```diff
@@ -177,7 +177,15 @@ def private_link(self):
 
     @classmethod
     def from_kwargs(cls, request_args, kwargs):
-        user = request_args.get('user') or kwargs.get('user') or _get_current_user()
+        from osf.models import ApiOAuth2PersonalToken
+        from framework.auth.cas import parse_auth_header
+
+        header_token = request.headers.get('Authorization', None)
```

Comment:
Drive-by: This will fail if `from_kwargs` is used outside of a request context. This *probably* isn't a blocker: `from_kwargs` *should* only be called within a request, but I'm not sure if that's true without doing a code dive. 

Ideally, we wouldn't be reading from the request at all here (read from the arguments instead), but I can see why you did it that way.

Reply:
Ah, just had a thought: what if you modified `_get_current_user` to check the bearer token instead of doing it here? That feels like a better place for this logic, since that function is already reading from the request.